File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        List<Map.Entry<Integer, Integer>> entryList = new ArrayList<>(map.entrySet());
        entryList.sort(Map.Entry.comparingByValue(Comparator.reverseOrder()));

        for (Map.Entry<Integer, Integer> entry : entryList) {
            Integer value = entry.getValue();
            if(k <= 0) {
                break;
            }
            answer++;
            k -= value;
        }

        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    fruit_count = {}

    for i in tangerine:
        fruit_count[i] = fruit_count.get(i, 0) + 1

    sorted_fruits = sorted(fruit_count.items(), key=lambda x: x[1], reverse=True)

    for fruit, count in sorted_fruits:
        if k <= 0:
            break
        answer += 1
        k -= count

    return answer